provider/gateway/router.go

```diff
@@ -116,9 +150,152 @@ func leaseServiceStatusHandler(log log.Logger, cclient cluster.ReadClient) http.
 }
 
 func leaseServiceLogsHandler(log log.Logger, cclient cluster.ReadClient) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-		http.Error(w, "unimplemented", http.StatusNotImplemented)
+	return func(w http.ResponseWriter, r *http.Request) {
+		service := mux.Vars(r)["serviceName"]
+		if service == "" {
+			http.Error(w, "invalid request", http.StatusBadRequest)
+			return
+		}
```

Comment:
let's handle this in the same way the other params are handled.  for better or for worse.